stats/cloud/collector.go

```diff
@@ -253,6 +253,12 @@ func (c *Collector) Run(ctx context.Context) {
 // Collect receives a set of samples. This method is never called concurrently, and only while
 // the context for Run() is valid, but should defer as much work as possible to Run().
 func (c *Collector) Collect(sampleContainers []stats.SampleContainer) {
+	select {
+	case <-c.stopSendingMetricsCh:
+		return
+	default:
+	}
+
```

Comment:
This `select` probably should be under the `if` below, though TBH I'm not 100% sure that `if` is actually necessary...

Reply:
I think regardless of their position, the functionality is the same. I place the select above because it looks like `If we stopped sending metrics already, return. Otherwise, execute Collect data logic`.

Reply:
Fair enough, and as you say, it doesn't really matter. In my mind "have we stopped sending metrics" is just another early return condition for a method that already had one, and since the `if referenceID... ` condition was here before it, we should put the new one after it.